api/V1/Device.js

```diff
@@ -82,32 +82,138 @@ module.exports = function(app, baseUrl) {
   /**
    * @api {get} /api/v1/devices Get list of devices
    * @apiName GetDevices
-   * @apiGroup Devices
+   * @apiGroup Device
    *
-   * @apiUse V1UserAuthorizationHeader
+   * @apiParam {JSON} where [Sequelize where conditions](http://docs.sequelizejs.com/manual/tutorial/querying.html#where) for query.
+   * @apiParam {Number} [userId] Get devices that belong to this user.
    *
    * @apiUse V1ResponseSuccess
-   * @apiSuccess {JSON} devices List of devices.
    *
    * @apiUse V1ResponseError
    */
   app.get(
-      apiUrl,
-      passport.authenticate(['jwt'], {session: false}),
-      async (request, response) => {
-          log.info(request.method + " Request: " + request.url);
+    apiUrl,
+    async (request, response) => {
+      log.info(request.method + " Request: " + request.url);
 
-          // Check that the request was authenticated by a User.
-          if (!requestUtil.isFromAUser(request))
-            return responseUtil.notFromAUser(response);
+      var where = request.query.where;
+      var queryUserId = request.query.userId;
+      try {
+        where = JSON.parse(where);
```

Comment:
I'm not crazy about accepting arbitrary where clauses from the API as it could be abused.

What do you actually need to be able to search by and could there just be a specific API for that use?

Reply:
How do you think this could be abused? Should we have it that you can only see devices that you have control of?

Reply:
Allowing arbitrary external input to be used as part of the query greatly increases the chances that someone will be able to get at data they're not supposed to be to. We're also then trusting that sequelize has no bugs which allow SQL injection attacks.

Allowing sequelize input also ties us to using sequelize. If we wanted to change how the API server accesses the database or be written in a different language we would have to break compatibility and figure out how to handle existing clients which use sequelize query. 

Details of the database layer shouldn't be leaking out to the API layer. It's safer & cleaner if we have specific query arguments for the things we need to expose over the API.